src/components/molecules/Visualizer/Engine/Cesium/hooks.ts

```diff
@@ -163,6 +191,206 @@ export default ({
+      !property?.cameraLimiter.cameraLimitterTargetArea ||
+      !geodsic
+    )
+      return undefined;
+    const targetHalfWidth = (property.cameraLimiter.cameraLimitterTargetWidth || targetWidth) / 2;
+    const targetHalfHeight = (property.cameraLimiter.cameraLimitterTargetLength || targetHeight) / 2;
+    const topDemention = geodsic.geodesicVertical.interpolateUsingSurfaceDistance(targetHalfHeight);
+    const bottomDemention = geodsic.geodesicVertical.interpolateUsingSurfaceDistance(-targetHalfHeight);
+    const rightDemention = geodsic.geodesicHorizontal.interpolateUsingSurfaceDistance(targetHalfWidth);
+    const leftDemention = geodsic.geodesicHorizontal.interpolateUsingSurfaceDistance(-targetHalfWidth);
+
+    const rightTop = new Cartographic(rightDemention.longitude, topDemention.latitude, 0);
+    const leftTop = new Cartographic(leftDemention.longitude, topDemention.latitude, 0);
+    const rightBottom = new Cartographic(rightDemention.longitude, bottomDemention.latitude, 0);
+    const leftBottom = new Cartographic(leftDemention.longitude, bottomDemention.latitude, 0);
+
+    return {
+      cartographicDimensions: {
+        rightDemention,
+        leftDemention,
+        topDemention,
+        bottomDemention,
+      },
+      cartesianArray: [
+        Cartographic.toCartesian(rightTop),
+        Cartographic.toCartesian(leftTop),
+        Cartographic.toCartesian(leftBottom),
+        Cartographic.toCartesian(rightBottom),
+        Cartographic.toCartesian(rightTop),
+      ],
+    };
+  }, [property?.cameraLimiter, geodsic]);
+
+  // calculate maximum camera view (outer boundaries)
+  const cameraViewOuterBoundaries = useMemo((): undefined | { cartesianArray: Cartesian3[] } => {
+    const viewer = cesium.current?.cesiumElement;
+    if (
+      !viewer ||
+      !viewer.scene ||
+      viewer.isDestroyed() ||
+      !property?.cameraLimiter ||
+      !property?.cameraLimiter.cameraLimitterTargetArea ||
+      !geodsic
+    )
+      return undefined;
+
+    const camera = new CesiumCamera(viewer.scene);
+    camera.setView({
+      destination: Cartesian3.fromDegrees(
+        property.cameraLimiter.cameraLimitterTargetArea.lng,
+        property.cameraLimiter.cameraLimitterTargetArea.lat,
+        property.cameraLimiter.cameraLimitterTargetArea.height,
+      ),
+      orientation: {
+        heading: property?.cameraLimiter?.cameraLimitterTargetArea.heading,
+        pitch: property?.cameraLimiter?.cameraLimitterTargetArea.pitch,
+        roll: property?.cameraLimiter?.cameraLimitterTargetArea.roll,
+        up: camera.up,
+      },
+    });
+
+    const computedViewRectangle = camera.computeViewRectangle() as Rectangle;
+    const rectangleHalfWidth = Rectangle.computeWidth(computedViewRectangle) * Math.PI * 1000000;
+    const rectangleHalfHeight = Rectangle.computeHeight(computedViewRectangle) * Math.PI * 1000000;
+
+    const targetHalfWidth = (property.cameraLimiter.cameraLimitterTargetWidth || targetWidth) / 2;
```

Comment:
You can shorten codes with

```ts
const {
  cameraLimitterTargetWidth: width,
  // ...
} = property?.cameraLimiter ?? {};
```

and use them

Reply:
@rot1024 , here I have to get value from object `property.cameraLimiter.cameraLimitterTargetWidth` or use default ones `targetWidth` and then divide by 2.
I think using the shorten version will add more code because I have to check `width || targetWidth`

Reply:
Isn't it enough?

```
const {
  cameraLimitterTargetWidth: width = targetWidth,
  // ...
} = property?.cameraLimiter ?? {};
```

Reply:
Thanks, I know `destructuring assignment` but didn't know you can assign default values to it.
Gonna use it.